MPF.Core/Utilities/Tools.cs

```diff
@@ -261,5 +291,315 @@ private static (string? tag, string? url) GetRemoteVersionAndUrl()
         }
 
         #endregion
+
+        #region PlayStation 3
+
+        /// <summary>
+        /// Validates a getkey log to check for presence of valid PS3 key
+        /// </summary>
+        /// <param name="logPath">Path to getkey log file</param>
+        /// <param name="key">Output 16 byte key, null if not valid</param>
+        /// <returns>True if path to log file contains valid key, false otherwise</returns>
+        public static bool ParseGetKeyLog(string? logPath, out byte[]? key, out byte[]? id, out byte[]? pic)
+        {
+            key = null;
+            id = null;
+            pic = null;
+
+            if (string.IsNullOrEmpty(logPath))
+                return false;
+
+            try
+            {
+                if (!File.Exists(logPath))
+                    return false;
+
+                // Protect from attempting to read from really long files
+                FileInfo logFile = new(logPath);
+                if (logFile.Length > 65536)
+                    return false;
+
+                // Read from .getkey.log file
+                using StreamReader sr = File.OpenText(logPath);
+
+                // Determine whether GetKey was successful
+                string? line;
+                while ((line = sr.ReadLine()) != null && line.Trim().StartsWith("get_dec_key succeeded!") == false) ;
+                if (line == null)
+                    return false;
+
+                // Look for Disc Key in log
+                while ((line = sr.ReadLine()) != null && line.Trim().StartsWith("disc_key = ") == false) ;
+                // If end of file reached, no key found
```

Comment:
Hate to ask this, but if you're putting comments between lines, can you have a newline before the comment? It may be worthwhile to combine some of the comments for logical chunks of code, like places that you do `null` or range checks.